fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -51,6 +51,33 @@ def build_verify_command(username, password, source = "/tmp", provider_short_nam
       not_implemented(__method__)
     end
 
+    # Runs preparations before executing any command from the executor.
+    #
+    # @param [Hash] api key containing the issuer id and private key
+    # @return [Hash] copy of `api_key` which includes an extra `key_dir` with the location of the .p8 file on disk
+    def prepare(original_api_key)
+      return if original_api_key.nil?
+      # Create .p8 file from api_key and provide api key info which contains .p8 file path
+      api_key = original_api_key.dup
+      if self.kind_of?(ShellScriptTransporterExecutor)
+        # as of Transporter v3.3.0, the app is unable to detect the private keys under the 'private_keys' folder in current directory
+        # so we must rely on the other search paths in the Home dir:
+        # https://help.apple.com/itc/transporteruserguide/en.lproj/static.html#itc803b7be80
+        private_keys_dir = File.join(Dir.home, ".appstoreconnect/private_keys")
```

Comment:
according to the [docs](https://help.apple.com/itc/transporteruserguide/en.lproj/static.html#apdA1aded1d2), a `private_keys` folder in the current directory SHOULD be supported (and would simplify the cleanup process a lot), but I couldn't make it work on my machine (bug in Transporter?), so I went ahead to use one of the other search paths suggested under $HOME